### PR TITLE
Updated Slider test 

### LINKED ISSUE #58120: Slider and RangeSlider Overlay painted twice
The Slider and RangeSlider overlay is painted twice.

The value indicator is added to the overlay when it is clicked or dragged because both of these gestures are occurring then it adds same value indicator twice into the overlay.

## Steps to Reproduce

Use Slider or RangeSlider. 

You will also see that when finding the Overlay in the sider test that the widget tester uses the method firstRenderObject(find.byType(Overlay)).

Instead of using firstRenderObject it should be using renderObject. The widget tester can't use renderObject on the Overlay since there are always two instances of the Overlay. So if it uses renderObject than it will return an error.

### PR DESCRIPTION
**Pull request by JoseAlba**
## Description

Updated test because Flutter is currently checking if there is an Overlay by using: tester.firstRenderObject(find.byType(Overlay)); instead of `tester.renderObject(find.byType(Overlay));`. We should always be checking if there is only one Overlay.

After updating test a few test were identifying that there were two Overlays. This was not because of the Slider but instead because of the testing. The test were using two widgets that were providing an `Overlay` context: MaterialApp and Navigator. To make the test better that were failing the Navigator widget had to be removed.

After removing that widget the test were no longer failing.

Also change the syntax a little bit for the Slider and RangeSlider for better readability and more consistency between the two files.
 
## Related Issues

Closes https://github.com/flutter/flutter/issues/58120

## Tests

I added the following tests: `slider.dart` and `range_slider.dart` test updated.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by rami-a — APPROVED** [review]

File: packages/flutter/test/material/range_slider_test.dart [hunk 1]
```diff
@@ -1409,11 +1398,14 @@ void main() {
...
       paints
-        ..rrect(color: sliderTheme.inactiveTrackColor)
-        ..rect(color: sliderTheme.activeTrackColor)
-        ..rrect(color: sliderTheme.inactiveTrackColor),
+        ..rrect(color: const Color(0x3d2196f3))
+        ..rect(color: const Color(0xff2196f3))
+        ..rrect(color: const Color(0x3d2196f3)),
     );
 
+    expect(valueIndicatorBox, paintsExactlyCountTimes(#drawRect, 1));
```
Comment (HansMuller) [inline packages/flutter/test/material/range_slider_test.dart:1406]: Please add a comment here that explains why we expect 1 drawRect and 2 drawRRects.

**Review by HansMuller — COMMENTED** [review]
Really glad to see this, the changes make the tests relatively straightforward. Just need a few trivial cleanups.

File: packages/flutter/lib/src/material/range_slider.dart [hunk 2]
```diff
@@ -1216,29 +1216,17 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
     _state.overlayController.reverse();
   }
 
-  void _handleDragStart(DragStartDetails details) {
-    _startInteraction(details.globalPosition);
-  }
+  void _handleDragStart(DragStartDetails details) => _startInteraction(details.globalPosition);
```
[the remark this answers is not in the document]
Comment (JoseAlba) [reply packages/flutter/lib/src/material/range_slider.dart:1219]: I think if we can't use the => notation is makes sense to have it as it used to be. I think having the separate lines are closer how the formatter works and easier for people to read.

File: packages/flutter/test/material/range_slider_test.dart [hunk 3]
```diff
@@ -1353,53 +1351,44 @@ void main() {
...
-                          ),
-                        );
-                      },
-                    ),
-                  ],
-                );
-              });
-            }),
+        home: Scaffold(
+          body: Builder(
```
[the remark this answers is not in the document]
Comment (JoseAlba) [reply packages/flutter/test/material/range_slider_test.dart:1355]: Done.

File: packages/flutter/test/material/range_slider_test.dart [hunk 4]
```diff
@@ -1353,53 +1351,44 @@ void main() {
...
+                  ),
+                  RaisedButton(
+                    child: const Text('Next'),
+                    onPressed: () {
+                      Navigator.of(context).pushReplacement(
+                        MaterialPageRoute<void>(
+                          builder: (BuildContext context) {
+                            return RaisedButton(
+                              child: const Text('Inner page'),
+                              onPressed: () => Navigator.of(context).pop(),
```
[the remark this answers is not in the document]
Comment (JoseAlba) [reply packages/flutter/test/material/range_slider_test.dart:1374]: Done

File: packages/flutter/test/material/range_slider_test.dart [hunk 5]
```diff
@@ -1409,11 +1398,14 @@ void main() {
     expect(
       valueIndicatorBox,
       paints
-        ..rrect(color: sliderTheme.inactiveTrackColor)
-        ..rect(color: sliderTheme.activeTrackColor)
-        ..rrect(color: sliderTheme.inactiveTrackColor),
+        ..rrect(color: const Color(0x3d2196f3))
```
[the remark this answers is not in the document]
Comment (JoseAlba) [reply packages/flutter/test/material/range_slider_test.dart:1401]: Done

File: packages/flutter/test/material/range_slider_test.dart (hunk 1 above)
Comment (JoseAlba) [reply packages/flutter/test/material/range_slider_test.dart]: Updated

**Comment by JoseAlba** [comment]
Updated all the test so that the overlay detects the value indicator path and paragraph.

There is an issue where the Overlay class sees everything in the screen. When testing the RangeSlider it does not contain the path and paragraph paint. Which means that the value indicator is not painted within the Slider or RangeSlider class.

**Pull request merged**